core/node/eth_watch/src/lib.rs

```diff
@@ -172,34 +132,60 @@ impl EthWatch {
         &mut self,
         storage: &mut Connection<'_, Core>,
     ) -> Result<(), EventProcessorError> {
-        let stage_latency = METRICS.poll_eth_node[&PollStage::Request].start();
-        let to_block = self.client.finalized_block_number().await?;
-        if to_block <= self.last_processed_ethereum_block {
-            return Ok(());
-        }
-
-        let events = self
-            .client
-            .get_events(
-                Web3BlockNumber::Number(self.last_processed_ethereum_block.into()),
-                Web3BlockNumber::Number(to_block.into()),
-                RETRY_LIMIT,
-            )
-            .await?;
-        stage_latency.observe();
-
         for processor in &mut self.event_processors {
-            let relevant_topic = processor.relevant_topic();
-            let processor_events = events
-                .iter()
-                .filter(|event| event.topics.first() == Some(&relevant_topic))
-                .cloned()
-                .collect();
-            processor
-                .process_events(storage, &*self.client, processor_events)
+            let client = match processor.event_source() {
+                EventsSource::L1 => self.l1_client.as_ref(),
+                EventsSource::SL => self.sl_client.as_ref(),
+            };
+            let chain_id = client.chain_id().await?;
+            let finalized_block = client.finalized_block_number().await?;
```

Comment:
nit: you can query `finalized_block` out of event_processors' loop

Reply:
In theory yes, but as the client can be different for different processors with current number of processors, it wouldn't have resulted in less queries (it's queried once for L1 and once for SL) and it would complicate code so I vote not to touch it